firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -900,6 +899,47 @@ private ListenerRegistration addSnapshotListenerInternal(
         firestore.getClient(), queryListener, activity, wrappedListener);
   }
 
+  @PublicApi
+  public String getPath() {
```

Comment:
Note to self: It's tempting to use a CollectionReference here but that only works for collection queries, not the soon-to-be-added "collection group" queries which are rooted at a document path.  Still, it would be nice to avoid exposing just a raw string path.